main.js

```diff
@@ -38,36 +38,36 @@ app.set('port', process.env.PORT || 3000);
 app.use(express.static('public'));
 
 // middleware
-app.use('/', router);
 app.use(
 	express.urlencoded({
 		extended: false,
-	}),
+	})
 );
 app.use(express.json());
 router.use(cookieParser('secret_passcode')); // this pass code is used for encrypting the cookie data 
 router.use(
 	expressSession({
 		secret: 'secret_passcode',
 		cookie: {
-			maxAge: 4000000,
+			maxAge: 4000000
 		},
 		resave: false,
 		saveUninitialized: false,
-	}),
+	})
 );
 router.use(connectFlash());
 // put flash message into response' local variable
 router.use((req, res, next) => {
-  res.locals.flashMessage = req.flash();
-  next();
+	res.locals.flashMessages = req.flash();
+	next();
 });
 router.use(
 	methodOverride('_method', {
 		methods: ['POST', 'GET'],
-	}),
+	})
 );
 // courses routes
+app.use('/', router);
```

Comment:
the order of this code is very important because you will get an error when you put this earlier than the codes related to flash messages.